### PR TITLE
Bump to blockifier `0.8.0-rc.2`, cairo `2.7.0`

### LINKED ISSUE #2368: Use non-rc 2.7.0

### PR DESCRIPTION
**Pull request by cptartur**
Closes #2368 

## Introduced changes

- Bumped to blockifier 0.8.0-rc.2
- Bumped to cairo 2.7.0
- Fixed broken tests
- Updated usage of `--snforge` flag in scarb

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/tests/e2e/declare.rs [hunk 1]
```diff
@@ -127,13 +127,13 @@ async fn test_happy_case_versions(version: &str) {
...
+#[test_case(Some("100000000000000000"), None, None; "max_fee")]
+#[test_case(None, Some("100000"), None; "max_gas")]
+#[test_case(None, None, Some("100000000000000"); "max_gas_unit_price")]
 #[test_case(None, None, None; "none")]
-#[test_case(Some("99999999999999999"), None, Some("999999999999"); "max_fee_max_gas_unit_price")]
-#[test_case(None, Some("9999"), Some("999999999999"); "max_gas_max_gas_unit_price")]
-#[test_case(Some("999999999999999"), Some("9999"), None; "max_fee_max_gas")]
+#[test_case(Some("10000000000000000000"), None, Some("100000000000000"); "max_fee_max_gas_unit_price")]
+#[test_case(None, Some("100000"), Some("100000000000000"); "max_gas_max_gas_unit_price")]
+#[test_case(Some("100000000000000000"), Some("100000"), None; "max_fee_max_gas")]
```
Comment (cptartur) [inline crates/sncast/tests/e2e/declare.rs:136]: Why were these changed?

Comment (cptartur) [reply crates/sncast/tests/e2e/declare.rs:136]: @integraledelebesgue

Comment (integraledelebesgue) [reply crates/sncast/tests/e2e/declare.rs:136]: When the prices changed, these tests became invalid. Also, it's more convenient to operate on powers of ten

**Review by integraledelebesgue — APPROVED** [review]
